src/buffer_control_block.cpp

```diff
@@ -42,8 +42,16 @@ bool BufferControlBlock::write(char *data, uint32_t size) {
 
 
 	int len = pwrite(BufferTree::backing_store, data, size, file_offset + storage_ptr);
-	if (len == -1) {
-		printf("ERROR: write to buffer %i failed %s\n", id, strerror(errno));
+	if (len == -1 || len != size) {
+		// try one more time
+		len = pwrite(BufferTree::backing_store, data, size, file_offset + storage_ptr);
```

Comment:
I'm not sure this is correct. If we write fewer than `size` bytes, we should only write the bytes that haven't been written already right?
maybe something like
`
while (len < size) {
  if (len == -1) {oops}
  else {size -= len; pwrite(size)} 
}
`

Reply:
Will do